dev/workflow/rocoto/gfs_tasks.py

```diff
@@ -2416,134 +2424,197 @@ def cleanup(self):
-                deps.append(rocoto.add_dependency(dep_dict))
+                deps_full.append(rocoto.add_dependency(dep_dict))
                 dep_dict = {'type': 'task', 'name': f'{self.run}_fbwind'}
-                deps.append(rocoto.add_dependency(dep_dict))
+                deps_full.append(rocoto.add_dependency(dep_dict))
 
                 if self.options['do_wave']:
                     dep_dict = {'type': 'task', 'name': f'{self.run}_waveawipsbulls'}
-                    deps.append(rocoto.add_dependency(dep_dict))
+                    deps_full.append(rocoto.add_dependency(dep_dict))
                     dep_dict = {'type': 'task', 'name': f'{self.run}_waveawipsgridded'}
-                    deps.append(rocoto.add_dependency(dep_dict))
+                    deps_full.append(rocoto.add_dependency(dep_dict))
 
             dep_dict = {'type': 'task', 'name': f'{self.run}_arch_vrfy'}
-            deps.append(rocoto.add_dependency(dep_dict))
+            deps_full.append(rocoto.add_dependency(dep_dict))
             if self.options['do_archcom']:
                 if self.options['do_globusarch']:
                     dep_dict = {'type': 'task', 'name': f'{self.run}_globus_arch'}
                 else:
                     dep_dict = {'type': 'metatask', 'name': f'{self.run}_arch_tars'}
 
-                deps.append(rocoto.add_dependency(dep_dict))
+                deps_full.append(rocoto.add_dependency(dep_dict))
 
-        dependencies = rocoto.create_dependency(dep_condition='and', dep=deps)
+        # Build the rocoto dependencies
+
+        deps_list = []
+        # Add half-cycle dependencies if they exist
+        if len(deps_half) > 1:
+            dependencies_half = rocoto.create_dependency(dep_condition='and', dep=deps_half)
+            dependencies_full = rocoto.create_dependency(dep_condition='and', dep=deps_full)
+            deps_list = [dependencies_half, dependencies_full]
+            # Combine half and full cycle dependencies with OR
+            deps_full_half = rocoto.create_dependency(dep_condition='or', dep=deps_list)
+
+        elif len(deps_half) == 1:
+            dependencies_half = rocoto.create_dependency(dep=deps_half)
+            dependencies_full = rocoto.create_dependency(dep_condition='and', dep=deps_full)
+            deps_list = [dependencies_half, dependencies_full]
+
+            # Combine half and full cycle dependencies with OR
+            deps_full_half = rocoto.create_dependency(dep_condition='or', dep=deps_list)
+
+        else:
+            deps_full_half = rocoto.create_dependency(dep=deps_full)
+
+        # Add full- and half-cycle to the complete list
+        deps_all.append(deps_full_half)
+
+        # Combine with AND
+        dependencies = rocoto.create_dependency(dep_condition='and', dep=deps_all)
 
         resources = self.get_resource('cleanup')
+        cycledef = self.run.replace('enkf', '')
+        cycledef = f'gdas_half, {cycledef}' if 'gdas' in cycledef else cycledef
```

Comment:
Is this if statement necessary? Removing this if statement allows for all cleanup tasks to reappear in my rocotostat job card. Turning on the gfs cycles also have the cleanup_jobs populate.
```suggestion
        cycledef = f'gdas_half, {cycledef}' 
```

Reply:
I tried again with the original if statement and now things are appearing like the should. This change is likely not needed and things can proceed.